homeassistant/components/xbox/base_sensor.py

```diff
@@ -52,12 +52,10 @@ def entity_picture(self) -> str:
         # to point to the correct image, with the correct domain and certificate
         # using YARL URL lib, we can replace the domain part of a url and append
         # only the queries we need (url and format) and omit unnecessary ones (padding)
-        original_url = URL(self.data.display_pic)
-        return str(
-            original_url.with_host("images-eds-ssl.xboxlive.com").with_query(
-                url=original_url.query["url"], format=original_url.query["format"]
-            )
-        )
+        url = URL(self.data.display_pic)
+        if url.host == "images-eds.xboxlive.com":
+            url = url.with_host("images-eds-ssl.xboxlive.com")
+        return str(url.with_query(url=url.query["url"], format=url.query["format"]))
```

Comment:
None of this is needed i think. You just need str(url). Everything else should have been retained.

Reply:
Also you seem to have lost the padding removal.

Reply:
I don't understand, you commented that I needed to check if the url needs replacing which I added.

This is the most elegant solution I could think of, because now it checks if the base url needs replacement and does that when necessary (row 56 and 57).

The on row 58, it returns the url with the query string which contains only the necessary queries url and format and thus removes the padding. 

How would you do this differently? You commented something like this before, I think I misunderstood then.

Reply:
The old code removed a padding url argument. The new doesn't.

Reply:
Okey you recreate with known url queries. But that is not equivalent.

It should have done something like.

with_query(url.query.pop("padding", None)

Reply:
Okay, i'll try that